src/test/java/com/mathworks/ci/systemTests/GlobalToolIT.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @Before
+    public void testSetup() throws IOException {
+        this.project = jenkins.createFreeStyleProject();
+        this.buildWrapper = new UseMatlabVersionBuildWrapper();
+        this.scriptBuilder = new RunMatlabCommandBuilder();
+    }
+
+    @After
+    public void testTearDown() {
+        this.project = null;
+        this.scriptBuilder = null;
+        Utilities.matlabInstDescriptor = null;
+        this.buildWrapper = null;
+    }
+
+    @Test
+    public void verifyGlobalToolForFreeStyle() throws Exception{
+        // Adding the MATLAB Global tool
+        Utilities.setMatlabInstallation("MATLAB_PATH", Utilities.getMatlabRoot(), jenkins);
+
+        // Selecting MATLAB that is defined as Global tool
+        MatlabInstallation _inst = MatlabInstallation.getInstallation("MATLAB_PATH");
+        MatlabBuildWrapperContent content = new MatlabBuildWrapperContent(_inst.getName(), null);
+        buildWrapper.setMatlabBuildWrapperContent(content);
+        project.getBuildWrappersList().add(buildWrapper);
+        scriptBuilder.setMatlabCommand("disp('apple')");
+        project.getBuildersList().add(scriptBuilder);
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        jenkins.assertLogContains(_inst.getHome(), build);
+        jenkins.assertBuildStatus(Result.SUCCESS, build);
+        jenkins.assertLogContains("apple", build);
+    }
+
+    @Test
+    public void verifyBuildFailsForIncorrectToolPath() throws Exception{
+        // Adding the MATLAB Global tool
+        Utilities.setMatlabInstallation("MATLAB_PATH", "matlab/incorrect/path", jenkins);
+
+        // Selecting MATLAB that is defined as Global tool
+        MatlabInstallation _inst = MatlabInstallation.getInstallation("MATLAB_PATH");
+        MatlabBuildWrapperContent content = new MatlabBuildWrapperContent(_inst.getName(), null);
+        buildWrapper.setMatlabBuildWrapperContent(content);
+        project.getBuildWrappersList().add(buildWrapper);
+
+        scriptBuilder.setMatlabCommand("ver");
+        project.getBuildersList().add(scriptBuilder);
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        jenkins.assertBuildStatus(Result.FAILURE, build);
+        jenkins.assertLogContains("Verify global tool configuration for the specified node.", build);
+        jenkins.assertLogContains("MATLAB_PATH", build);
+    }
+
+    @Test
+    public void verifyAllToolsAreAvailable() throws Exception {
+        // Adding the Global tools
+        String matlabRoot = System.getenv("MATLAB_ROOT");
+        String matlabRoot22b = System.getenv("MATLAB_ROOT_22b");
+        Assume.assumeTrue("Not running tests as MATLAB_ROOT_22b environment variable is not defined", matlabRoot22b != null && !matlabRoot22b.isEmpty());
+
+        Utilities.setMatlabInstallation("MATLAB_PATH_1", matlabRoot, jenkins);
+        Utilities.setMatlabInstallation("MATLAB_PATH_2", matlabRoot22b, jenkins);
```

Comment:
Same question here. What's setting `MATLAB_ROOT_22b` environment variable?

Reply:
So the idea is to run the matrix tests against two MATLAB releases. One release is the latest automatically picked by setup MATLAB action. Second release is 2022b(no particular reason as such to chose this particular release)

Reply:
Got it, thanks